make devnet-up is broken

<!--
Need help?
Refer to our contributing guidelines for additional information about making a good issue:
https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->


**Bug Description**
<!--
A clear and concise description of what the bug is.
-->

`make devnet-up`
results in this error:
```
Traceback (most recent call last):
  File "/Users/samlaf/devel/eigenlayer/da-integrations/optimism/./bedrock-devnet/main.py", line 9, in <module>
    main()
  File "/Users/samlaf/devel/eigenlayer/da-integrations/optimism/./bedrock-devnet/main.py", line 5, in main
    devnet.main()
  File "/Users/samlaf/devel/eigenlayer/da-integrations/optimism/bedrock-devnet/devnet/__init__.py", line 120, in main
    devnet_deploy(paths)
  File "/Users/samlaf/devel/eigenlayer/da-integrations/optimism/bedrock-devnet/devnet/__init__.py", line 250, in devnet_deploy
    l2_output_oracle = addresses['L2OutputOracleProxy']
                       ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'L2OutputOracleProxy'
make: *** [devnet-up] Error 1
```

**Environment Information:**
<!--
 - Operating System: [e.g. Ubuntu 20.04]
 - Package Version (or commit hash): [e.g. op-node v1.5.1, op-node c934019]
-->
on develop branch 

**Configurations:**
<!--
Command line flags or environment variables you're using.
-->

**Logs:**
<!--
Logs and/or error messages that help illustrate the issue.
-->

**Additional context**


Think this is the culprit https://github.com/ethereum-optimism/optimism/pull/11817/files#diff-d5b1923bb59d7a16b2affecb73b46f01db70d432a231d3ef5db751c4a2e42c5bL703: seems like the saving of the output oracle to addresses.json file was removed. Guessing this is just a mistake and we need to add it back, but not sure what this new interface is supposed to do. Maybe this is intended and its __init__.py which needs to be changes to remove this reading of L2outputoracleproxy?

--- a/bedrock-devnet/devnet/__init__.py
+++ b/bedrock-devnet/devnet/__init__.py
@@ -247,11 +247,7 @@ def devnet_deploy(paths):
     wait_for_rpc_server('127.0.0.1:9545')
 
     # Print out the addresses being used for easier debugging.
-    l2_output_oracle = addresses['L2OutputOracleProxy']
-    dispute_game_factory = addresses['DisputeGameFactoryProxy']
     batch_inbox_address = rollup_config['batch_inbox_address']
-    log.info(f'Using L2OutputOracle {l2_output_oracle}')
-    log.info(f'Using DisputeGameFactory {dispute_game_factory}')
     log.info(f'Using batch inbox {batch_inbox_address}')
 
     # Set up the base docker environment.
@@ -264,7 +260,11 @@ def devnet_deploy(paths):
     # Must be done selectively because op-proposer throws if both are set.
     if DEVNET_L2OO:
         docker_env['L2OO_ADDRESS'] = l2_output_oracle
+        l2_output_oracle = addresses['L2OutputOracleProxy']
+        log.info(f'Using L2OutputOracle {l2_output_oracle}')
     else:
+        dispute_game_factory = addresses['DisputeGameFactoryProxy']
+        log.info(f'Using DisputeGameFactory {dispute_game_factory}')
         docker_env['DGF_ADDRESS'] = dispute_game_factory
         docker_env['DG_TYPE'] = '254'
         docker_env['PROPOSAL_INTERVAL'] = '12s'

